### ISSUE TITLE
use DruidDataSource Connection happen “ java.sql.SQLFeatureNotSupportedException: null ”

### ISSUE THREAD

**Issue by github-zhoukun** [housepower/ClickHouse-Native-JDBC#249]
### Environment

- OS version: mac 10.14.6 
- JDK version: jdk1.8
- ClickHouse Server version: 20.3.6.40
- ClickHouse Native JDBC version:  2.4.2
- (Optional) Spark version: N/A
- (Optional) Other components' version: N/A
- alibaba druid version:  1.2.3

### Error logs

```
java.sql.SQLFeatureNotSupportedException: null
	at com.github.housepower.jdbc.wrapper.SQLConnection.getCatalog(SQLConnection.java:102) ~[clickhouse-native-jdbc-shaded-2.4.2.jar:?]
	at com.alibaba.druid.pool.DruidPooledConnection.getCatalog(DruidPooledConnection.java:927) ~[druid-1.2.3.jar:1.2.3]
	at com.alibaba.druid.pool.DruidPooledConnection.prepareStatement(DruidPooledConnection.java:341) ~[druid-1.2.3.jar:1.2.3]
	at com.zk.flinkplugin.clickhouse.BatchingOutputOperate.flush(BatchingOutputOperate.java:100) ~[classes/:?]
	at com.zk.flinkplugin.clickhouse.BatchingOutputOperate.lambda$open$0(BatchingOutputOperate.java:67) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_261]
	at java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:308) [?:1.8.0_261]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java) [?:1.8.0_261]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_261]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_261]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_261]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_261]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_261]
```

#### Happen Exception code 
sql = "INSERT INTO mergetree_table values(?,?,?)"
```
 try (Connection connection = this.dataSource.getConnection()) {
                    try (PreparedStatement ps = connection.prepareStatement(this.sql)) {
...
}
```

#### Modified to no problem code
```
  try (Connection connection = this.dataSource.getConnection().getConnection()) {
                    try (PreparedStatement ps = connection.prepareStatement(this.sql)) { ....}
```

When the connection is DruidPooledConnection happen Exception!

**Comment by github-zhoukun** [comment]
bug submit

**Comment by pan3793** [comment]
Thx for report issue. This feature has been implemented in mainline code, will back port to 2.4.x later.
